Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/CompeteController.cs

```diff
@@ -381,24 +431,28 @@ public ActionResult SubmitBinaryFile(BinarySubmissionModel participantSubmission
                 throw new HttpException((int)HttpStatusCode.BadRequest, Resource.ContestsGeneral.Upload_file);
             }
 
-            var problem = this.Data.Problems.All().FirstOrDefault(x => x.Id == participantSubmission.ProblemId);
+            var problem = this.problemsData.GetById(participantSubmission.ProblemId);
             if (problem == null)
             {
                 throw new HttpException((int)HttpStatusCode.Unauthorized, Resource.ContestsGeneral.Problem_not_found);
             }
 
-            if (official && !this.ValidateContestIp(this.Request.UserHostAddress, problem.ContestId))
-            {
-                return this.RedirectToAction("NewContestIp", new { id = problem.ContestId });
-            }
-
-            var participant = this.Data.Participants.GetWithContest(problem.ContestId, this.UserProfile.Id, official);
+            var participant = this.participantsData
+                .GetWithContestByContestIdUserIdAndIsOfficial(problem.ContestId, this.UserProfile.Id, official);
             if (participant == null)
             {
                 throw new HttpException((int)HttpStatusCode.Unauthorized, Resource.ContestsGeneral.User_is_not_registered_for_exam);
             }
 
             this.ValidateContest(participant.Contest, official);
+
+            if (official &&
+                !participant.Contest.IsOnline &&
```

Comment:
Why do we exclude Online contests from Ip validation, the fact that they'll probably be opened, does not mean that no IP validation should happen.